packages/flutter/test/rendering/paragraph_test.dart

```diff
@@ -375,6 +375,27 @@ void main() {
     expect(paragraph.size.height, 30.0);
   }, skip: isBrowser); // https://github.com/flutter/flutter/issues/61018
 
+  test('textAlign triggers TextPainter relayout in the paint method', () {
+    final RenderParagraph paragraph = RenderParagraph(
+      const TextSpan(text: 'A', style: TextStyle(fontSize: 10.0)),
+      textDirection: TextDirection.ltr,
+      textAlign: TextAlign.left,
+    );
+
+    Rect getRectForA() => paragraph.getBoxesForSelection(const TextSelection(baseOffset: 0, extentOffset: 1)).single.toRect();
+
+    layout(paragraph, constraints: const BoxConstraints(maxWidth: 100.0));
+
+    expect(getRectForA(), const Rect.fromLTWH(0, 0, 10, 10));
+
+    paragraph.textAlign = TextAlign.right;
+    expect(paragraph.debugNeedsLayout, isFalse);
+    expect(paragraph.debugNeedsPaint, isTrue);
+
+    paragraph.paint(MockPaintingContext(), Offset.zero);
+    expect(getRectForA(), const Rect.fromLTWH(0, 0, 10, 10));
+  });
+
```

Comment:
Do we have enough existing tests to cover that when you change the configuration of a RenderParagraph or RenderEditable that the new configuration is properly accounted for in the intrinsics calculations?

Reply:
Added a few tests.  I couldn't think of ways to test some configurations using the test font (`textDirection` for example).